beetsplug/web/__init__.py

```diff
@@ -218,6 +226,16 @@ def item_query(queries):
     return g.lib.items(queries)
 
 
+@app.route('/item/path/<everything:path>')
+def item_at_path(path):
+    query = beets.library.PathQuery('path', path.encode('utf-8'))
```

Comment:
Nice! I like this solution—it gobbles up the entire remainder of the URL, which makes it safe to use slashes wherever necessary. Woohoo!